src/groups/z_bmq/z_bmqa/z_bmqa_confirmeventbuilder.cpp

```diff
@@ -0,0 +1,67 @@
+#include <z_bmqa_confirmeventbuilder.h>
+#include <bmqa_confirmeventbuilder.h>
+
+
+int z_bmqa_ConfirmEventBuilder__delete(z_bmqa_ConfirmEventBuilder const** builder_obj) {
+    using namespace BloombergLP;
+
+    const bmqa::ConfirmEventBuilder* builder_p = reinterpret_cast<const bmqa::ConfirmEventBuilder*>(*builder_obj);
+    delete builder_p;
```

Comment:
In general we like to use these `BSLS_ASSERT` macros to help check the contract of functions. In this case, we should note in the docstring for this function where undefined behavior exists, and insert these assert macros to check against it.

```suggestion
    BSLS_ASSERT(builder_obj != NULL);

    const bmqa::ConfirmEventBuilder* builder_p = reinterpret_cast<const bmqa::ConfirmEventBuilder*>(*builder_obj);
    delete builder_p;
```